### DISCUSSION TITLE
A Possible New Way to Share Files Between Trilium for Mac & Linux

### DISCUSSION THREAD

**Discussion by blynn357** [TriliumNext/Trilium#4521] [Q&A]
I have installed Trilium on my Mac Mini running the most recent Mac OS and on Linux Mint. The primary hard drive on my Mac has a partition formatted as ExFAT. My idea is to keep the trilium-data folder on this Ex-FAT partition. I used a similar concept about 20 years ago when I was running Windows and Linux Xandros. I was using Thunderbird for my email. There was just POP for email at the time so I created a shared partition for the Thunderbird email files and it worked great.

My question is can Trilium run on Mac OS and Linux share data? 

The next question is what do I need to do to point Trilium to the trilium-data file on the aforementioned Shared drive. I know that sharing has worked for other programs that run on both Mac and Linux.

**Comment by blynn357** [comment]
I found something posted on this site in reference to exporting data. It suggested: 
export TRILIUM_DATA_DIR=/home/myuser/data/my-trilium-data. 
I modified it as follows which did not work:
export /Sonoma/Users/bruce/Library/Application\ Support/trilium-data=/Shared/trilium-data.
The comment when the command failed was: export: not valid in this context.

**Reply by blynn357** [reply]
Ignore the above.

 I found the directions here for doing this on a Mac: https://github.com/zadam/trilium/wiki/Data-directory.

I received the following error message:
Uncaught Exception:
Error: ENOTDIR: not a directory, mkdir '/Users/bruce/Library/LaunchAgents/set.trilium.env.plist/log'
at Object.mkdirSync (node:fs:1396:3)
at Object.<anonymous> (/Applications/Trilium Notes.app/Contents/Resources/app.asar/src/services/log.js:8:8)
at Module._compile (node:internal/modules/cjs/loader:1269:14)
at Module._extensions..js (node:internal/modules/cjs/loader:1324:10)
at Module.load (node:internal/modules/cjs/loader:1124:32)
at Module._load (node:internal/modules/cjs/loader:965:12)
at f._load (node:electron/js2c/asar_bundle:2:13377)
at Module.require (node:internal/modules/cjs/loader:1148:19)
at require (node:internal/modules/cjs/helpers:110:18)
at Object.<anonymous> (/Applications/Trilium Notes.app/Contents/Resources/app.asar/src/services/sql_init.js:1:13)

I feel like I am getting very close but not being a programmer the above is not something I can comprehend.

**Comment by Nriver** [comment]
You get it wrong.

Environment variable can be set as `export name=value`.

 `TRILIUM_DATA_DIR` is the `name` of the environment variable. You should change the `value`.

What you want is probably this:
```
export TRILIUM_DATA_DIR=/Sonoma/Users/bruce/Library/Application\ Support/trilium-data
```

**Comment by blynn357** [comment]
Thank you Nriver. I realized after my last post that I was following the directions for accomplishing this in Linux. Right now I am trying to get Trilium to work on my Mac with the Trilium data directory, trilium-data, residing on my Shared partition formatted as ExFAT. Here are the instructions given in this wiki. I followed these and received the above error message.

Mac OS X

You need to create a .plist file under ~/Library/LaunchAgents to load it properly each login.

To load it manually, u need to use launchctl setenv TRILIUM_DATA_DIR <yourpath>

Here is a pre-defined template, where you just need to add your path to:

<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">

<plist version="1.0">
    <dict>
        <key>Label</key>
        <string>set.trilium.env</string>
        <key>RunAtLoad</key>
        <true/>
        <key>ProgramArguments</key>
        <array>
            <string>launchctl</string>
            <string>setenv</string>
            <string>TRILIUM_DATA_DIR</string>
            <string>/Users/YourUserName/Library/Application Support/trilium-data</string>
        </array>
    </dict>
</plist>
